app/code/Magento/Customer/Model/Options.php

```diff
@@ -97,12 +97,14 @@ private function prepareNamePrefixSuffixOptions($options, $isOptional = false)
         if (empty($options)) {
             return false;
         }
+
         $result = [];
-        $options = array_filter(explode(';', $options));
+        $options = explode(';', $options);
         foreach ($options as $value) {
-            $value = $this->escaper->escapeHtml(trim($value));
+            $value = $this->escaper->escapeHtml(trim($value)) ?: ' ';
             $result[$value] = $value;
         }
+
         if ($isOptional && trim(current($options))) {
             $result = array_merge([' ' => ' '], $result);
```

Comment:
Well, I don't understand the logic after such change. In optional field we add empty value, but in required field we allow empty value. How empty value can be valid for a required field?

Does it really works this way in registration?

https://github.com/magento/magento2/issues/18823#issuecomment-433606725